src/python/pants/binaries/executable_pex_tool.py

```diff
@@ -9,10 +9,14 @@
 from pex.pex_info import PexInfo
 
 from pants.backend.python.subsystems.pex_build_util import PexBuilderWrapper
+from pants.base.deprecated import deprecated_module
 from pants.subsystem.subsystem import Subsystem
 from pants.util.dirutil import is_executable, safe_concurrent_creation
 
 
+deprecated_module('1.19.0.dev0', 'Use PythonToolBase and PythonToolPrepBase instead!')
```

Comment:
When would I use one vs the other? Are these 1-1 equal, and all I have to do is change the parent class, or will I have to update any of the class first?

Reply:
`ExecutablePexTool` is a mishmash attempt at doing `PythonTool*` as well as allowing non-static requirements (which is what `ResolveRequirementsTaskBase` does, at least). Given benjy's comment below we can probably avoid this hint message altogether.